plugins/destination/s3/client/write.go

```diff
@@ -66,10 +66,31 @@ func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data
 	return nil
 }
 
-// sanitizeJSONKeys replaces all invalid characters in JSON keys with underscores.
-// It does the replacement in-place, modifying the original object. This is required
+// sanitizeRecordJSONKeys replaces all invalid characters in JSON keys with underscores. This is required
 // for compatibility with Athena.
-func sanitizeJSONKeys(obj any) {
+func sanitizeRecordJSONKeys(record arrow.Record) arrow.Record {
+	cols := make([]arrow.Array, record.NumCols())
+	for i, col := range record.Columns() {
+		if arrow.TypeEqual(col.DataType(), types.NewJSONType()) {
+			b := types.NewJSONBuilder(array.NewExtensionBuilder(memory.DefaultAllocator, types.NewJSONType()))
+			for r := 0; r < int(record.NumRows()); r++ {
+				if col.IsNull(r) {
+					b.AppendNull()
+					continue
+				}
+				obj := col.GetOneForMarshal(r)
+				sanitizeJSONKeysForObject(obj)
+				b.Append(obj)
+			}
+			cols[i] = b.NewArray()
+			continue
+		}
+		cols[i] = col
```

Comment:
personal preference
```suggestion
		if !arrow.TypeEqual(col.DataType(), types.NewJSONType()) {
			cols[i] = col
			continue
		}

		b := types.NewJSONBuilder(array.NewExtensionBuilder(memory.DefaultAllocator, types.NewJSONType()))
		for r := 0; r < int(record.NumRows()); r++ {
			if col.IsNull(r) || !col.IsValid(r) {
				b.AppendNull()
				continue
			}

			obj := col.GetOneForMarshal(r)
			sanitizeJSONKeysForObject(obj)
			b.Append(obj)
		}

		cols[i] = b.NewArray()
	}		
```

Reply:
Yeah, maybe slightly better but I think it's good enough as it is

Reply:
100%, just wanted to use exit-early (continue) on the shorter path